### PR TITLE
Fix side condition for BITVECTOR_MULT.

### PR DESCRIPTION
**Pull request by mpreiner**
Add additonal tracing for side conditions.

### CODE REVIEW

File: src/theory/quantifiers/bv_inverter.cpp [hunk 1]
```diff
@@ -452,18 +458,24 @@ Node BvInverter::solve_bv_lit(Node sv,
           /* with side condition:
            * ctz(t) >= ctz(s) <-> x * s = t
            * where
-           * ctz(t) >= ctz(s) -> (t & -t) >= (s & -s)  */
+           * ctz(t) >= ctz(s) -> (t & -t) >= (s & -s) /\ s != 0 */
```
Comment (aniemetz) [inline src/theory/quantifiers/bv_inverter.cpp]: This is not entirely correct. If `s == 0` and `t == 0` then any value for x is inverse. But if `s == 0` and `t != 0`, then there exists no inverse.

This should be:
`... /\ (s == 0 => t == 0)`

Comment (aniemetz) [reply src/theory/quantifiers/bv_inverter.cpp]: On second thought, the above is still not correct. Also, the main problem is that `0 & -0 = 0`, hence, if `s == 0` or `t == 0` and `s != t` our encoding for `ctz(t) >= ctz(s)` is incorrect. Seems that we have to be careful to cover all zero cases.

After checking with CVC4 (without cbqi-bv), this should be the correct SC:
`t == 0 \/ ((t & -t) >= (s & -s) /\ (s == 0 => t == 0))`

I checked both directions, i.e., the following implications:
`not (exists x. s * x = t  =>  SC)`  ... if sat, SC is too restrictive
`not (SC => exists x. s * x = t)`     ... if sat SC is too weak

Comment (ajreynol) [reply src/theory/quantifiers/bv_inverter.cpp]: Sounds right to me, to simplify what you wrote:
`t == 0 \/ ((t & -t) >= (s & -s) /\ s != 0)`

Comment (mpreiner) [reply src/theory/quantifiers/bv_inverter.cpp]: Done.

**Pull request merged**